operator_ui/src/pages/Jobs/RegionalNav.tsx

```diff
@@ -252,9 +254,12 @@ const RegionalNavComponent = ({
                 )}
                 {definition && (
                   <Button
+                    onClick={() =>
+                      setPersistJobSpec(JSON.stringify(definition))
+                    }
                     href={{
```

Comment:
```suggestion
                    href={`/jobs/new?definition=${JSON.stringify(definition)}`}
```

Benefits of this:
- You don't need to use `onClick` on a link, which is scary.
- You can parse the definition to figure out if it's TOML or JSON and set the correct value in the job page.
- No need to use any storage.